### PR TITLE
Add new channels for community participation

### PR DESCRIPTION
**Pull request by weierophinney**
This pull request adds links and information on new community participation channels, including:

- the new Zend Framework Slack
- the newsletter

Updates were made to the contributor guide page, which is where IRC was previously messaged (it no longer is), as well as to the footer (to link to the slack and the newsletter signup pages).

### CODE REVIEW

**Comment by PowerKiKi** [comment]
Have you considered alternatives to Slack, such as Gitter ?

I feel Slack does not promote openness as much as it could. Discussions are hidden behind a login. Meaning they are not indexed by Google. A user need a to create and manage a different account for each community ("team" in Slack parlance). And the Slack itself is not open-source.

All those points are solved by Gitter. It also provides integration with almost 30 services, including of course GitHub and Travis. Moreover they [announced a few weeks ago](http://blog.gitter.im/2017/03/15/gitter-gitlab-acquisition/) that they were going 100% open-source.

What is your opinion about that ?

**Comment by AxaliaN** [comment]
I like Slack, mainly because I am used to it though. We have been using it for a year at work now, no real problems. Only, like PowerKiKi said, it is not open.

File: templates/app/contributor-guide.phtml [hunk 1]
```diff
@@ -61,26 +61,52 @@
...
+        <ul>
+            <li><a href="https://zendframework.slack.com/messages/C4QBQUEG5">#general</a> channel (general discussion)</li>
+            <li><a href="https://zendframework.slack.com/messages/C4R13GTRD">#contributors</a> channel (discussions relating to the development of Zend Framework itself)</li>
+            <li><a href="https://zendframework.slack.com/messages/C4RF4G6LE">#apigility</a> channel (discussion)</li>
+            <li><a href="https://zendframework.slack.com/messages/C4RF4P0A2">#apigility-contrib</a> channel (contributor discussion)</li>
+            <li><a href="https://zendframework.slack.com/messages/C4RF57B0E">#expressive</a> channel (discussion)</li>
+            <li><a href="https://zendframework.slack.com/messages/C4QRP8ELU">#expressive-contrib</a> channel (contributor discussion)</li>
+        </ul>
+    </li>
+    <li><a href="https://zendframework.slackarchive.com">Archives</a></li>
```
Comment (xtreamwayz) [inline templates/app/contributor-guide.phtml]: Should be: https://zendframework.slackarchive.io/

**Comment by weierophinney** [comment]
> Have you considered alternatives to Slack, 

We did. We decided on slack due to existing ecosystem, familiarity amongst a broad set of users, and amount of support available. We have solved the issue of search engine indexing by integrating with slackarchive (thanks for pointing out the typo, @xtreamwayz ). And we _want_ login requirements, to better allow moderation (something we have struggled with on IRC for years now). It's a good fit for the goals we have.

**Comment by PowerKiKi** [comment]
Thanks for explaining the rationale.

I still feel that requiring a login to read is counter-productive, even though I can see how it makes sense if it is to write.

Anyway I guess it's too late to convince you otherwise.

**Pull request merged**